man/tilestache-server.1
.TH TILESTACHE-SERVER 1 "Jan 24, 2011"
.SH NAME
tilestache-server \- seed a single layer in your TileStache configuration
.SH SYNOPSIS
.B tilestache-server
.RI [ options ]
.SH DESCRIPTION
This manual page documents briefly the \fBtilestache-server\fR command.
.PP
\fBtilestache-server\fR serves tiles through Werkzeug, using a
tilestache configuration file.
.SH OPTIONS
.TP
\fB\-c\fR, \fB\-\-config\fR \fIfile\fR
Path to configuration file. Defaults to \fItilestache.cfg\fR.
.TP
\fB-i\fR, \fB\-\-ip\fR \fIaddress\fR
IP address to listen to. Defaults to \fI127.0.0.1\fR.
.TP
\fB-p\fR, \fB\-\-port\fR \fIport\fR
Port number to listen to. Defaults to \fI8080\fR.
.TP
\fB\-\-include\-path\fR \fIpaths\fR
Add the following colon-separated list of paths to Python's include path (aka sys.path).
.TP
.B \-h, \-\-help
Show summary of options.
.SH SEE ALSO
.BR tilestache-render (1)
.BR tilestache-seed (1)
.SH AUTHOR
\fBTileStache\fR was written by Michal Migurski <mike@stamen.com>.
.PP
This manual page was written by David Paleino <dapal@debian.org>,
for the Debian project (and may be used by others).


CHANGELOG
2011-11-23: 1.19.4
- Fixed a silly bug where I was comparing ints and strings.

2011-11-23: 1.19.3
- Merged llimllib's support for zoom limits in Composite provider.

2011-10-09: 1.19.2
- Added support for GCPs and cubicspline in GDAL provider.

2011-10-06: 1.19.1
- Fixed TypeError bug with Access-Control-Allow-Origin in WSGITileServer.

2011-09-04: 1.19.0
- Added new resampling option to GDAL provider in Goodies.

2011-08-30: 1.18.3
- Fixed incorrect handling of grayscale datasources in GDAL provider.

2011-08-24: 1.18.2
- Fixed alpha channel handling for mixed-alpha combinations in Composite provider.

2011-07-23: 1.18.1
- Fixed new spacing configuration option on vector provider.

2011-07-23: 1.18.0
- Added new configuration options to vector provider: clipped=padded for Polymaps and spacing.

2011-07-18: 1.17.0
- Added configuration for HTTP Access-Control-Allow-Origin response header for cross-origin resource sharing (CORS).
- Added --include-path option to tilestache-server.py.

2011-07-15: 1.16.0
- Added configuration for JPEG and PNG creation options, including 8-bit PNG output and Photoshop color table support.

2011-07-13: 1.15.0
- Added three new options to tilestache seed: --from-mbtiles, --enable-retries and --error-list.

2011-07-12: 1.14.0
- Added per-layer "write cache" option.
- Added --tile-list option to tilestache-seed.py, so that tiles to seed can be explicitly enumerated.
- Fixed S3 cache provider to work with boto 1.9 and 2.0 equally.
- Added the simplest GDAL provider that could possibly work.

2011-07-04: 1.13.0
- Added support for MBTiles format. Tilesets can be providers or destinations for seeded tiles.

2011-06-22: 1.12.1
- Added new adjustments and blend mode to Composite provider: threshold, curves2, and linear light.

2011-06-16: 1.12.0
- Changed argument structure of osm2pgsql in MirrorOSM provider to allow kept coastlines.

2011-06-14: 1.11.1
- Stopped using wasteful boto.s3.Connection.create_bucket() call in S3 cache provider.
- Mega-update from Dane Springmeyer with fixes to examples, documentation, and projections.

2011-05-13: 1.11.0
- Added ignore-cache parameter to tile rendering; renamed from Zain's cache-bust parameter.

2011-04-18: 1.10.6
- Added faint mapnik tile image to MirrorOSM image responses.

2011-04-18: 1.10.5
- Fixed Grid provider font path bug, again.

2011-04-18: 1.10.4
- Fixed Grid provider font path bug.

2011-04-15: 1.10.3
- Fixed cache lifespan flub in TileStache.Caches.Disk.

2011-04-14: 1.10.2
- Fixed TileStache.Vector in setup.py.

2011-04-13: 1.10.1
- Fixed cache lifespan flub in Memcache.

2011-04-13: 1.10.0
- Added new caches: Multi, Memcache, and S3.
- Added several new Vector response types: ESRI GeoServices JSON, ESRI AMF, and BSON encoding.
- Added new MirrorOSM provider for populating PostGIS with piecemeal OpenStreetMap data.
- Added "cache lifespan" to core Layer configuration.

2011-04-08: 1.9.1
- Bug fix from kaolin: Geography.py was missing import of ModestMaps.Geo Location

2011-03-28: 1.9.0
- Added new Vector provider based on old PostGeoJSON provider.

2011-03-28: 1.8.1
- Merged Zain Memon's proxy provider update.

2011-03-24: 1.8.0
- Added new Core.NoTileLeftBehind exception class to suppress result caching.

2011-03-17: 1.7.4
- Fixed use of "layers.get(name)" to "name in layers" in WSGI server.
- Made possible to build Composite docs without numpy or sympy.

2011-03-17: 1.7.3
- Made it possible to instantiate WSGI server with a configuration object instead of a file.

2011-01-31: 1.7.2
- Fixed json import to account for older Python versions, by checking for simplejson availability.

2011-01-29: 1.7.1
- Fixed PIL import to account for systems that can "import Image".

2011-01-26: 1.7.0
- Made geographic projections user-specifiable using module:class naming.
- Made all class specifications accept module:class syntax, hoping to deprecate module.class sometime.
- Added manpage fixes from David Paleino.

2011-01-26: 1.6.2
- Fixed WSGI content-length type error that was affecting mod_wsgi.

2011-01-21: 1.6.0
- Added "-d" flag for tilestache-seed.py to drop tiles in a directory, regardless of cache settings.
- Merged Zain Memon's WSGI support, supercedesprevious example gunicorn support.

2010-11-29: 1.5.0
- Expanded PostGeoJSON provider in extras collection to include built-in shape clipping for Polymaps.
- Added gzip option to Disk cache to reduce disk usage of  text-based providers like XML or JSON.

2010-11-23: 1.4.1
- Being permissive of I/O errors when getting sublayers in Composite provider.
  This is short-term necessary, may require configuration or fixing later.

2010-11-20: 1.4.0
- Completed substantial upgrades to Composite provider (TileStache.Goodies.Providers.Composite),
  including multiple blend modes, curves adjustments, and layer opacity.

2010-11-18: 1.3.0
- Added new URL Template provider from Ian Dees to TileStache.Providers.

2010-11-15: 1.2.2
- Made it possible to use just a subset of layer preview arguments instead of the full four (lat, lon, zoom, ext).

2010-11-15: 1.2.1
- Previously neglected to include VERSION file in tarball.

2010-11-15: 1.2.0
- Made preview.html pages for tiles configurable with a starting location.
- Sharply improved TileStache.Goodies.Providers.Composite with JSON config, etc. 

2010-11-09: 1.1.3
- Found another mapnik bug, removed unnecessary tempfile creation.

2010-11-09: 1.1.2
- Refixed a bug in mapnik provider and learned to test things before pushing them.

2010-11-09: 1.1.1
- Fixed a bug in mapnik provider to make fonts directory work.

2010-10-21: 1.1.0
- Made configuration objects customizable by defining properties in new API.html.


scripts/tilestache-server.py
#!/usr/bin/env python
"""tilestache-server.py will serve your cache.

This script is intended to be run directly from the command line.

It is intended for direct use only during development or for debugging TileStache.

For the proper way to configure TileStach for serving tiles see the docs at:

http://tilestache.org/doc/#serving-tiles

To use this built-in server, install werkzeug and then run tilestache-server.py:

    tilestache-server.py

By default the script looks for a config file named tilestache.cfg in the current directory and then serves tiles on http://127.0.0.1:8080/. 

You can then open your browser and view a url like:

    http://localhost:8080/osm/0/0/0.png

The above layer of 'osm' (defined in the tilestache.cfg) will display an OpenStreetMap
tile proxied from http://tile.osm.org/0/0/0.png
   
Check tilestache-server.py --help to change these defaults.
"""

if __name__ == '__main__':
    from datetime import datetime
    from optparse import OptionParser, OptionValueError
    import os, sys

    parser = OptionParser()
    parser.add_option("-c", "--config", dest="file", default="tilestache.cfg",
        help="the path to the tilestache config")
    parser.add_option("-i", "--ip", dest="ip", default="127.0.0.1",
        help="the IP address to listen on")
    parser.add_option("-p", "--port", dest="port", type="int", default=8080,
        help="the port number to listen on")
    parser.add_option('--include-path', dest='include',
        help="Add the following colon-separated list of paths to Python's include path (aka sys.path)")
    (options, args) = parser.parse_args()

    if options.include:
        for p in options.include.split(':'):
            sys.path.insert(0, p)

    from werkzeug.serving import run_simple
    import TileStache

    if not os.path.exists(options.file):
        print >> sys.stderr, "Config file not found. Use -c to pick a tilestache config file."
        sys.exit(1)

    app = TileStache.WSGITileServer(config=options.file, autoreload=True)
    run_simple(options.ip, options.port, app)

